File: Algorithms/086.PartitionList.cpp
class Solution {
public:
    ListNode *partition(ListNode *head, int x) {
        ListNode* leftheadptr = new ListNode(0);
        ListNode* rightheadptr = new ListNode(0);
        ListNode* left = leftheadptr;
        ListNode* right = rightheadptr;
        while (head != NULL) {
            if (head->val < x) {
                left->next = head;
                left = left->next;
            } else {
                right->next = head;
                right = right->next;
            }
            head = head->next;
        }
        left->next = rightheadptr->next;
        right->next = NULL;
        return leftheadptr->next;
    }
};

File: Algorithms/086.PartitionList.py
from leetcode import *
class Solution:
    # @param head, a ListNode
    # @param x, an integer
    # @return a ListNode
    def partition(self, head, x):
        leftheadptr = ListNode(0)
        rightheadptr = ListNode(0)
        left = leftheadptr
        right = rightheadptr
        while head:
            if head.val < x:
                left.next = head
                left = left.next
            else:
                right.next = head
                right = right.next
            head = head.next
        left.next = rightheadptr.next
        right.next = None
        return leftheadptr.next
